build-release.sh

```diff
@@ -330,12 +330,13 @@ if [ "${build_mono}" == "1" ]; then
   mkdir -p Godot_mono.app/Contents/MacOS
   cp out/macosx/tools-mono/godot.osx.opt.tools.universal.mono Godot_mono.app/Contents/MacOS/Godot
   mkdir -p Godot_mono.app/Contents/{Frameworks,Resources}
-  mkdir -p Godot_mono.app/Contents/{Frameworks,Resources}/GodotSharp
-  mkdir -p Godot_mono.app/Contents/{Frameworks,Resources}/GodotSharp/Mono
-  cp -rp out/macosx/tools-mono/GodotSharp/Api Godot_mono.app/Contents/Frameworks/GodotSharp
-  cp -rp out/macosx/tools-mono/GodotSharp/Mono/lib Godot_mono.app/Contents/Frameworks/GodotSharp/Mono
-  cp -rp out/macosx/tools-mono/GodotSharp/Tools Godot_mono.app/Contents/Frameworks/GodotSharp
-  cp -rp out/macosx/tools-mono/GodotSharp/Mono/etc Godot_mono.app/Contents/Resources/GodotSharp/Mono
+  mkdir -p Godot_mono.app/Contents/Resources/GodotSharp
+  mkdir -p Godot_mono.app/Contents/Frameworks/Mono
+  cp -rp out/macosx/tools-mono/GodotSharp Godot_mono.app/Contents/Resources
+  mv Godot_mono.app/Contents/Resources/GodotSharp/Mono/lib/*.dylib Godot_mono.app/Contents/Frameworks/Mono
+  # cp /usr/local/lib/libMoltenVK.dylib Godot_mono.app/Contents/Frameworks
+  sed -i -e 's/$mono_libdir/$mono_libdir\/..\/..\/..\/..\/Frameworks\/Mono/g' Godot_mono.app/Contents/Resources/GodotSharp/Mono/etc/mono/config
```

Comment:
I think it will also do the same thing with macOS sed. Usually I've done a check and made a `$SED` variable, but was trying to avoid having yet-another-complication added here, since these scripts are run in Linux containers. If we want to make sure it's more portable, though, that's an easy bit of logic to add.